include/net/buf.h

```diff
@@ -19,6 +19,12 @@
 extern "C" {
 #endif
 
+/**
+ * @brief Network buffer library
+ * @defgroup net_buf Network Buffer Library
```

Comment:
This one is supposed to show up in the generated docs, so it also needs a reference in doc/api/networking.rst for the generated API docs to show up:

     Network Buffer
     ==============

     .. doxygengroup:: net_buf
        :project: Zephyr
        :content-only:

Same goes for other API information for files below (that are supposed to be public interfaces).